lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm

```diff
@@ -36,6 +39,13 @@ sub _getDevices {
 
     my $root   = $params{root};
 
+    my %map = (
+        'MODEL'        => [ \&_getHdparmInfo, \&getInfoFromSmartctl ],
+        'MANUFACTURER' => [ \&getInfoFromSmartctl ],
+        'WWN'          => [ \&_getHdparmInfo ],
+    );
+    $map{DISKSIZE} = $map{FIRMWARE} = $map{DESCRIPTION} = $map{MODEL};
+
```

Comment:
It seems a comment on that code was lost in translation... Can I suggest you to better use the following code ?
```suggestion
    my $hdparm_and_smartctl_sources = [ \&_getHdparmInfo, \&getInfoFromSmartctl ];
    my $hdparm_sources   = [ \&_getHdparmInfo ];
    my $smartctl_sources = [ \&getInfoFromSmartctl ];

    my %sources = (
        MODEL           => $hdparm_and_smartctl_sources,
        DISKSIZE        => $hdparm_and_smartctl_sources,
        FIRMWARE        => $hdparm_and_smartctl_sources,
        DESCRIPTION     => $hdparm_and_smartctl_sources,
        MANUFACTURER    => $smartctl_sources,
        WWN             => $hdparm_sources,
    );
```
and adapt your code to use %sources ? This is more explicit than %map as we are searching a value at different sources.

Reply:
I don't quite like this. What if another source is added, and another?

Reply:
At least we see clearly what can be impacted. With your code, the problem is if the map for MODEL is changed, you must not miss this will impact DISKSIZE, FIRMWARE & DESCRIPTION as the big equality is not really intuitive.

But you may try another approach using such list:
```suggestion
    my %map = qw(
        MODEL         hdparm,smartctl
        MANUFACTURER  smartctl
        WWN           hdparm
        DISKSIZE      hdparm,smartctl
        FIRMWARE      hdparm,smartctl
        DESCRIPTION   hdparm,smartctl
    );
```
And update the loop to split $map{field} into $sub and explicitely doing the expected call to find $info{$sub}.

If you want callbacks, you still can have another list for callbacks:
```perl
    my %cb = (
        hdparm   => \&_getHdparmInfo,
        smartctl => \&getInfoFromSmartctl,
    );
```

Reply:
Let's just declare all subs explicitly.

```perl
    my %map = (
        'MODEL'        => [ \&_getHdparmInfo, \&getInfoFromSmartctl ],
        'DISKSIZE'     => [ \&_getHdparmInfo, \&getInfoFromSmartctl ],
        'FIRMWARE'     => [ \&_getHdparmInfo, \&getInfoFromSmartctl ],
        'DESCRIPTION'  => [ \&_getHdparmInfo, \&getInfoFromSmartctl ],
        'MANUFACTURER' => [ \&getInfoFromSmartctl ],
        'WWN'          => [ \&_getHdparmInfo ],
    );
```